src/functional/program_processors/codegens/gtfn/gtfn_backend.py

```diff
@@ -18,24 +18,30 @@
 import functional.iterator.ir as itir
 from eve import codegen
 from functional.iterator.transforms.pass_manager import apply_common_transforms
-from functional.program_processors.codegens.gtfn.codegen import GTFNCodegen
+from functional.program_processors.codegens.gtfn.codegen import GTFNCodegen, GTFNIMCodegen
+from functional.program_processors.codegens.gtfn.gtfn_ir_to_gtfn_im_ir import GTFN_IM_lowering
 from functional.program_processors.codegens.gtfn.itir_to_gtfn_ir import GTFN_lowering
 
 
 def generate(program: itir.FencilDefinition, **kwargs: Any) -> str:
     transformed = program
     offset_provider = kwargs.get("offset_provider")
+    do_unroll = not ("imperative" in kwargs and kwargs["imperative"])
```

Comment:
I'm not completely sure I follow. I was under the assumption that there should be no lifts left once we get to `gtfn_ir` or am I mistaken? The test case `fvm_nabla` leads to a situation where a reduction is "outlined" by the cse pass, a situation my backend explicitly doesn't handle. 

I just noticed that the imperative backend rejects unrolled reductions (by mistake). It does indeed handle them with a somewhat graceful fallback, consider e.g.:

```      
auto red_2 = [=](auto _step_3) {
        return _step_3(_step_3(_step_3(0., 0_c), 1_c), 2_c);
      }([=](auto _acc_1, auto _i_2) {
        return (_acc_1 + (gtfn::deref(gtfn::shift(kh_smag_ec, C2E, _i_2)) *
                          gtfn::tuple_get(_i_2, gtfn::deref(e_bln_c_s))));
      });
      auto tupl_1_0 = (red_2 / gtfn::deref(diff_multfac_smag));
      auto red_3 = [=](auto _step_6) {
        return _step_6(_step_6(_step_6(0., 0_c), 1_c), 2_c);
      }([=](auto _acc_4, auto _i_5) {
        return (_acc_4 + (gtfn::deref(gtfn::shift(vn, C2E, _i_5)) *
                          gtfn::tuple_get(_i_5, gtfn::deref(geofac_div))));
      });
      auto tupl_1_1 = red_3;
      auto tupl_1 = gtfn::make_tuple(tupl_1_0, tupl_1_1);
      return tupl_1;
```

I'll need to fix the compatibility check to that end

Reply:
I think what I meant was, maybe we have cases where the inline_lift fails, because we didn't unroll. In that case I was wondering if we can just fallback to do the unroll.